ReactDOM page translation (#48)
reactjs/pt-BR.reactjs.org: content/docs/reference-react-dom.md

permalink: docs/react-dom.html
---

Se você carregar o React através de uma tag `<script>`, essas APIs de nível superior estão disponíveis globalmente em `ReactDOM`. Se você usa ES6 com npm, você pode escrever `import ReactDOM from 'react-dom'`. Se você usa ES5 com npm, você pode escrever `var ReactDOM = require('react-dom')`.

## Visão Geral {#overview}

> O pacote `react-dom` provê métodos específicos para o DOM que podem ser usado no nível superior de sua aplicação como uma válvula de escape para sair do modelo do React se você precisar. A maioria de seus componentes não deverá precisar deste módulo.

Reviewer: ```suggestion
O pacote `react-dom` provê métodos específicos para o DOM que podem ser usados no nível superior de sua aplicação como uma válvula de escape para sair do modelo do React se você precisar. A maioria de seus componentes não devem precisar deste módulo.
```

- [`render()`](#render)
- [`hydrate()`](#hydrate)
- [`unmountComponentAtNode()`](#unmountcomponentatnode)
- [`findDOMNode()`](#finddomnode)
- [`createPortal()`](#createportal)

### Suporte dos Navegadores {#browser-support}

O React suporta todos os navegadores populares, incluindo Internet Explorer 9 e acima, apesar de [precisar de alguns _polyfills_](/docs/javascript-environment-requirements.html) para navegadores antigos como IE 9 e IE 10.

> Nota
>

> > Nós não suportamos navegadores antigos que não suportam métodos ES5, mas você pode descobrir que suas aplicações funcionam em navegadores antigos se _polyfills_ como [es5-shim and es5-sham](https://github.com/es-shims/es5-shim) forem incluídos na página. Você está por sua conta se optar por seguir esse caminho.

Reviewer 2: ```suggestion
Nós não temos suporte para navegadores antigos que não possuem suporte para métodos ES5. Mas, você pode descobrir que suas aplicações funcionam em navegadores antigos se _polyfills_ como [es5-shim and es5-sham](https://github.com/es-shims/es5-shim) forem incluídos na página. Você estará colocando a sua conta em risco se optar por seguir esse caminho.
```

Author: Não seria melhor "Você estará por sua conta e risco"? Acho que é a forma mais comum de usar essa expressão

Se a _callback_ opcional for fornecida, ela será executada depois do componente ser renderizado ou atualizado.

> Nota:
>
> `ReactDOM.render()` controla o conteúdo do nó contêiner que você passa. Qualquer elemento do DOM que existe dentro será substituído na primeira chamada. As próximas chamadas usam o algoritmo de diferenciação do React para atualizações eficientes.
>
> `ReactDOM.render()` não modifica o nó contêiner (só modifica os filhos do contêiner). Pode ser possível inserir um componente em um nó já existente no DOM sem sobrescrever os filhos existentes.
>
> `ReactDOM.render()` atualmente retorna uma referência à instância raiz de `ReactComponent`. Porém, usar esse valor de retorno é legado
> e deve ser evitado porque versões futuras do React podem renderizar componentes assincronamente em alguns casos. Se você precisa de uma referência da instância raiz de `ReactComponent`, a solução preferida é de anexar uma
> [callback de ref](/docs/more-about-refs.html#the-ref-callback-attribute) para o elemento raiz.
>

> > Usar `ReactDOM.render()` para fazer hidratar em um contêiner renderizado no servidor está deprecado e será removido no React 17. Ao invés disso, use [`hydrate()`](#hydrate).

Reviewer: ```suggestion
> Usar `ReactDOM.render()` para hidratar um contêiner renderizado no servidor está deprecado e será removido no React 17. Ao invés disso, use [`hydrate()`](#hydrate).
```
